### PR TITLE
chore: CS - apply `phpdoc_types_order.sort_algorithm=alpha`

### PR DESCRIPTION
**Pull request by keradus**
| Q             | A
| ------------- | ---
| Branch?       | 7.1 
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Issues        | Fix CS 
| License       | MIT

I suggest to unify types in phpdoc, for combined types (eg `foo|bar`).

So far:
- put `null` as last
- everything else as is (random order?)

Idea:
- put `null` as last
- everything else sorted alphabetically

### CODE REVIEW

**Comment by xabbuh** [comment]
I do not really care about the order. That said, if I would start a new codebase I would probably sort types alphabetically, but also put classes before scalar types and `false` and `null` at the end. But I do not have any hard feelings about how we handle this in the Symfony codebase.

**Comment by keradus** [comment]
I believe it's better to have some sorting than none (kind of `none = sort:random`).
right now, we have only alpha sorting possibility (with null at end).

for any more sophisticated sorting, one would need to implement ;)

File: src/Symfony/Component/Serializer/Normalizer/NormalizerInterface.php [hunk 1]
```diff
@@ -59,7 +59,7 @@ public function supportsNormalization(mixed $data, string $format = null, array
      * Use type "object" to match any classes or interfaces,
      * and type "*" to match any types.
      *
-     * @return array<class-string|'*'|'object'|string, bool|null>
+     * @return array<'*'|'object'|class-string|string, bool|null>
```
Comment (smnandre) [inline src/Symfony/Component/Serializer/Normalizer/NormalizerInterface.php:62]: I'd let "phpstan-like types" first

File: src/Symfony/Component/Config/Definition/Builder/NormalizationBuilder.php [hunk 2]
```diff
@@ -45,7 +45,7 @@ public function remap(string $key, string $plural = null): static
     /**
      * Registers a closure to run before the normalization or an expression builder to build it if null is provided.
      *
-     * @return ExprBuilder|$this
+     * @return $this|ExprBuilder
```
Comment (smnandre) [inline src/Symfony/Component/Config/Definition/Builder/NormalizationBuilder.php:48]: I'd force "$this" / self / etc at right :|

File: src/Symfony/Component/Process/Process.php [hunk 3]
```diff
@@ -1123,7 +1123,7 @@ public function getInput()
      *
      * This content will be passed to the underlying process standard input.
      *
-     * @param string|resource|\Traversable|self|null $input The content
+     * @param resource|self|string|\Traversable|null $input The content
```
Comment (smnandre) [inline src/Symfony/Component/Process/Process.php:1126]: Same, could "self"  have a special rule ... ? :)

File: src/Symfony/Component/Serializer/Normalizer/NormalizerInterface.php (hunk 1 above)
Comment (keradus) [reply src/Symfony/Component/Serializer/Normalizer/NormalizerInterface.php:62]: Currently, Fixer does not support such logic.
Feel free to introduce it via [raising PR](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/compare)

File: src/Symfony/Component/Config/Definition/Builder/NormalizationBuilder.php (hunk 2 above)
Comment (keradus) [reply src/Symfony/Component/Config/Definition/Builder/NormalizationBuilder.php:48]: Currently, Fixer does not support such logic.
Feel free to introduce it via [raising PR](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/compare)

File: src/Symfony/Component/Process/Process.php (hunk 3 above)
Comment (keradus) [reply src/Symfony/Component/Process/Process.php:1126]: Currently, Fixer does not support such logic.
Feel free to introduce it via [raising PR](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/compare)

**Comment by nicolas-grekas** [comment]
Reading the comments here, I think we might not want to do this change.
I don't think the current order is random - instead, I'd say each method author wrote it in the way that made sense to them.
And I see that alpha ordering doesn't make sense to many. I don't know if there is some non-alpha order that we could all agree on. Eg native types, then FQCN, then literal types, then null - but what about complex types, etc. I can't decide myself.

Better close and do nothing IMHO as I also see a lot of potential for bikeshadding here ;)

**Comment by chalasr** [comment]
I agree with Nicolas. Thanks for proposing.

**Pull request closed without merging**